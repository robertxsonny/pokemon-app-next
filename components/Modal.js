import styled from '@emotion/styled';
import { sm } from '../styles/breakpoints';
import { Emoji } from '../styles/shared';

const ModalContainer = styled.div({
  backgroundColor: 'rgba(0,0,0,0.5)',
  position: 'fixed',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  zIndex: 999
})

const ModalContent = styled.div({
  backgroundColor: 'white',
  width: '100%',
  height: 'auto',
  maxWidth: 320,
  maxHeight: 480,
  margin: 48,
  padding: 24,
  boxSizing: 'border-box',
  borderRadius: 6,
  fontSize: 13,
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  [sm.down]: {
    margin: 24
  }
})

const ModalTitle = styled.h3({
  fontSize: 16,
  marginTop: 0,
  textAlign: 'center'
})

const Modal = ({ children, open, title, emoji }) => {
  if (!open) {
    return null;
  }

  return (
    <ModalContainer>
      <ModalContent data-testid="modal">
        {emoji && <Emoji role="img" aria-label={title}>{emoji}</Emoji>}
        {title && <ModalTitle>{title}</ModalTitle>}
        {children}
      </ModalContent>
    </ModalContainer>
  )
}

export default Modal;