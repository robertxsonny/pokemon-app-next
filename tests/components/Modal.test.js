import { render } from '@testing-library/react';
import Modal from '../../components/Modal';

describe('Modal component', () => {
  test('Should render modal if open', () => {
    const { container } = render(<Modal open>Modal Content Goes Here</Modal>);
    expect(container).toMatchSnapshot();
  })

  test('Should render modal with title if open', () => {
    const { container } = render(<Modal open title="Modal Title">Modal Content Goes Here</Modal>);
    expect(container).toMatchSnapshot();
  })

  test('Should render modal with emoji if open', () => {
    const { container } = render(<Modal open emoji="ðŸ˜ƒ" title="Modal Title">Modal Content Goes Here</Modal>);
    expect(container).toMatchSnapshot();
  })

  test('Should not render modal if not open', () => {
    const { container } = render(<Modal open={false}>Modal Content Goes Here</Modal>);
    expect(container).toMatchSnapshot();
  })
})