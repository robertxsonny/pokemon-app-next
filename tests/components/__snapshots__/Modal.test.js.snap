// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Should not render modal if not open 1`] = `<div />`;

exports[`Modal component Should render modal if open 1`] = `
<div>
  <div
    class="css-1f6dfq1-ModalContainer e2hvhql2"
  >
    <div
      class="css-pujlrx-ModalContent e2hvhql1"
    >
      Modal Content Goes Here
    </div>
  </div>
</div>
`;

exports[`Modal component Should render modal with emoji if open 1`] = `
<div>
  <div
    class="css-1f6dfq1-ModalContainer e2hvhql2"
  >
    <div
      class="css-pujlrx-ModalContent e2hvhql1"
    >
      <span
        aria-label="Modal Title"
        class="css-1k4msm7-Emoji e55skjj7"
        role="img"
      >
        ðŸ˜ƒ
      </span>
      <h3
        class="css-1f0is5a-ModalTitle e2hvhql0"
      >
        Modal Title
      </h3>
      Modal Content Goes Here
    </div>
  </div>
</div>
`;

exports[`Modal component Should render modal with title if open 1`] = `
<div>
  <div
    class="css-1f6dfq1-ModalContainer e2hvhql2"
  >
    <div
      class="css-pujlrx-ModalContent e2hvhql1"
    >
      <h3
        class="css-1f0is5a-ModalTitle e2hvhql0"
      >
        Modal Title
      </h3>
      Modal Content Goes Here
    </div>
  </div>
</div>
`;
